### DISCUSSION TITLE
What would be the typical data access rates?

### DISCUSSION THREAD

**Discussion by brad-4k** [valhalla/valhalla#3993] [Q&A]
Thank you again for all of the hard work.  We have a nice little AWS setup running Valhalla.  We've been running it for some time but ran into a strange issue.

The system was running close to max on the disk (EFS) access rates.  We made a configuration change and restarted the instances and the utilization dropped down to about 10% (vs 80% to 90%).  All appears to be running ok but it has me wondering if there is a rule of thumb for access rates versus requests?

Say I have something like 100/s requests with a median route length of 300 km, what might I expect (roughly) the read rate from the dataset?  I assume that 200 Mb/s would be considerably high?

**Comment by kevinkreiser** [comment]
hard to say. the os has a disk cache which will prioritize certain parts of the disk (probably most frequently accessed ones) and depending on your configuration it may need to swap stuff in and out of that cache as requests come up. in my personal experience the os will fill ram with disk cache (for the memory mapped tar file). in my personal experience EFS has been too slow and ive always relied on gp2/3 nvme ebs volumes. EFS takes a big latency hit when disk cache needs to page something in from (network attached) file.

im not exactly sure what you mean by utilization when refering to EFS so im not sure if ive answered or properly understood your question but hopefully the above has at least some useful information in it. basic gist is at the end of the day important data needs to be in ram. disk is just a place to pull stuff into ram so reducing the latency of that read from disk into ram is important. setting the disk up as network attached storage will be a hinderance to and disk  cache misses because that read through network will be slower than local disk.

**Reply by brad-4k** [reply]
Latency isn't so much of an issue, it's more the throughput.  I was seeing data rates hitting 250 Mb/s which, at first blush, looks excessive.  After a reset, it's more like 15 to 25 Mb/s.  I'm trying to gauge which would be more typical for the kinds of routing requests I described in the post.
